docs/source/advanced/profiler.rst

```diff
@@ -3,10 +3,291 @@
+    |  on_train_batch_end                              |  7.6117e-05     	|  7.6117e-05     |
+    |  run_training_epoch                              |  0.0036915      	|  0.0036915      |
+    |  [LightningModule]BoringModel.on_epoch_end       |  1.079e-06      	|  1.079e-06      |
+    |  [LightningModule]BoringModel.on_train_end       |  1.509e-06      	|  1.509e-06      |
+    |  [LightningModule]BoringModel.on_fit_end         |  3.127e-06      	|  3.127e-06      |
+    |  [LightningModule]BoringModel.teardown           |  2.581e-06      	|  2.581e-06      |
+    -----------------------------------------------------------------------------------------------
+
+
+Advanced Profiler
+=================
+
+If you want more information on the functions called during each event, you can use the :class:`~pytorch_lighting.profiler.advanced.AdvancedProfiler`.
+This option uses Python's `cProfiler <https://docs.python.org/3/library/profile.html#module-cProfile>`_ to provide a report of time spent on *each* function called within your code.
+
+.. code-block:: python
+
+    # by passing a string
+    trainer = Trainer(..., profiler="advanced")
+
+    # or by passing an instance
+    profiler = AdvancedProfiler()
+    trainer = Trainer(..., profiler=profiler)
+
+The profiler's results will be printed at the completion of ``trainer.fit()``. This profiler
+report can be quite long, so you can also specify a ``dirpath`` and ``filename`` to save the report instead
+of logging it to the output in your terminal. The output below shows the profiling for the action
+``get_train_batch``.
+
+.. code-block::
+
+    Profiler Report
+
+    Profile stats for: get_train_batch
+            4869394 function calls (4863767 primitive calls) in 18.893 seconds
+    Ordered by: cumulative time
+    List reduced from 76 to 10 due to restriction <10>
+    ncalls  tottime  percall  cumtime  percall filename:lineno(function)
+    3752/1876    0.011    0.000   18.887    0.010 {built-in method builtins.next}
+        1876     0.008    0.000   18.877    0.010 dataloader.py:344(__next__)
+        1876     0.074    0.000   18.869    0.010 dataloader.py:383(_next_data)
+        1875     0.012    0.000   18.721    0.010 fetch.py:42(fetch)
+        1875     0.084    0.000   18.290    0.010 fetch.py:44(<listcomp>)
+        60000    1.759    0.000   18.206    0.000 mnist.py:80(__getitem__)
+        60000    0.267    0.000   13.022    0.000 transforms.py:68(__call__)
+        60000    0.182    0.000    7.020    0.000 transforms.py:93(__call__)
+        60000    1.651    0.000    6.839    0.000 functional.py:42(to_tensor)
+        60000    0.260    0.000    5.734    0.000 transforms.py:167(__call__)
+
+
+PyTorch Profiler
+================
+
+Autograd includes a profiler that lets you inspect the cost of different operators
+inside your model - both on the CPU and GPU. It uses the built-in :class:`~pytorch_lightning.profiler.pytorch.PyTorchProfiler`.
```

Comment:
Let's add a link to their repo.

Reply:
The profiler doc link is mentioned in 2 lines below. Is there any other reference you are referring to?